### PR TITLE
Harden ancillary `exec` function

### LINKED ISSUE #1273: cml pr escape chars
`touch 'foo(bar).baz' && cml pr create .` fails with `Syntax error: \"(\" unexpected`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Closes #1273

### CODE REVIEW

File: src/drivers/github.js [hunk 1]
```diff
@@ -273,15 +273,19 @@ class Github {
...
+        await this.runnerToken(),
+        '--url',
+        this.repo,
+        '--name',
+        name,
+        '--labels',
+        labels,
+        '--work',
+        resolve(workdir, '_work'),
+        ...(single ? ['--ephemeral'] : [])
```
Comment (dacbd) [inline src/drivers/github.js:288]: please include a quick note for future readers/contributors

```suggestion
        // note: '...' unpacking on an empty array to achieve a no-op effect
        ...(single ? ['--ephemeral'] : [])
```

**Review by dacbd — COMMENTED** [review]
LGTM rubber stamp (haven't ran it myself)

File: src/drivers/github.js (hunk 1 above)
Comment (0x2b3bfa0) [reply src/drivers/github.js:288]:
```suggestion
        // adds `--ephemeral` to the array only if `single` is set
        ...(single && ['--ephemeral'])
```

**Review by casperdcl — COMMENTED** [review]
Great to change `foo ${bar}` to `'foo', bar`.
But seems a bit excessive to change all instances of `'foo bar'` to `'foo', 'bar'`.

File: src/cml.js [hunk 2]
```diff
@@ -504,14 +504,23 @@ class CML {
...
-        if ((await exec('git rev-parse --is-shallow-repository')) === 'true') {
-          return await exec('git fetch --all --unshallow');
+        if (
+          (await exec('git', 'rev-parse', '--is-shallow-repository')) === 'true'
+        ) {
+          return await exec('git', 'fetch', '--all', '--unshallow');
         }
       } else {
-        return await exec(`git fetch --all --depth=${fetchDepth}`);
+        return await exec('git', 'fetch', '--all', `--depth=${fetchDepth}`);
```
[the remark this answers is not in the document]
Comment (0x2b3bfa0) [reply src/cml.js]: Resolved with https://github.com/iterative/cml/pull/1275/commits/dbba86b24f9a95c4f02087db98c8c4eda7c9d7fc

**Comment by 0x2b3bfa0** [comment]
> _But seems a bit excessive to change all instances of 'foo bar' to 'foo', 'bar'._

Do you have in mind any better approach?

File: src/drivers/github.js (hunk 1 above)
Comment (dacbd) [reply src/drivers/github.js]: I would say its use for object unpacking is more common than for arrays. 

Your linked example is already barf js code, and I would not consider it an excuse to continue. Consider that in your linked code, this is the norm, but here in this new chunk, it is out of place. Thus a quick footnote would be nice.

I don't understand your allergic reaction to code comments.

side note I think that `['thing', ...(false && ['maybe thing']) ]` works

File: src/drivers/github.js [hunk 3]
```diff
@@ -273,15 +273,20 @@ class Github {
...
+        '--url',
+        this.repo,
+        '--name',
+        name,
+        '--labels',
+        labels,
+        '--work',
+        resolve(workdir, '_work'),
+        // adds `--ephemeral` to the array only if `single` is set
+        ...(single && ['--ephemeral'])
```
Comment (dacbd) [inline src/drivers/github.js:289]: I don't think that `['thing', ...(false && ['maybe thing']) ]` works

Comment (dacbd) [reply src/drivers/github.js:289]:
```
node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> ['thing', ...(false && ['maybe thing']) ]
Uncaught TypeError: (false && ["maybe thing"]) is not iterable
>
```
?

Comment (0x2b3bfa0) [reply src/drivers/github.js]: Yes, major blunder; I was too lazy, fixed with https://github.com/iterative/cml/pull/1275/commits/b89b383eec5d89be0d3bd8bb7441bb2415973384

File: src/drivers/github.js (hunk 1 above)
Comment (0x2b3bfa0) [reply src/drivers/github.js:289]: > _Your linked example is already barf js code, and I would not consider it an excuse to continue._

Speaking of which, if there is any more idiomatic syntax, I'd be glad to change it. It's not my cup of tea either.

> _Consider that in your linked code, this is the norm, but here in this new chunk, it is out of place. Thus a quick footnote would be nice._

That's why I committed a comment, with the _footnote_. 🦶🏼🔫 

> _I don't understand your allergic reaction to code comments._

Sorry, it's probably the season. You know, pollen, dust, https://github.com/iterative/leo-server/pull/26#pullrequestreview-1204495281, ... 😄 

> _side note I **don't** think that `['thing', ...(false && ['maybe thing']) ]` works_

You're right, moved to https://github.com/iterative/cml/pull/1275#discussion_r1040339366

**Review by dacbd — APPROVED** [review]
:rubber_stamp:

**Pull request merged**